src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -40,8 +108,8 @@ private void InitializeCancellationToken(TimeSpan timeout)
             }
             else if (timeout > TimeSpan.Zero)
             {
-                _cts = new CancellationTokenSource(timeout);
-                _cancellationToken = _cts.Token;
+                _cts = new CancellationTokenSource();
+                _cancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_cts.Token, TimeoutTokenSource.FromTimeout((int)timeout.TotalMilliseconds)).Token;
```

Comment:
I was able to further cut down another allocation by making the registered delegate static. There seem to be a throughput improvement of around 3%.